src/main/java/io/vertx/core/http/impl/HttpServerImpl.java

```diff
@@ -638,14 +639,12 @@ protected void handleMessage(ServerConnection conn, ContextImpl context, Channel
           case BINARY:
           case CONTINUATION:
           case TEXT:
+          case PONG:
             conn.handleMessage(msg);
             break;
           case PING:
             // Echo back the content of the PING frame as PONG frame as specified in RFC 6455 Section 5.5.2
-            ch.writeAndFlush(new WebSocketFrameImpl(FrameType.PONG, wsFrame.getBinaryData()));
-            break;
-          case PONG:
-            // Just ignore it
+            chctx.channel().writeAndFlush(new PongWebSocketFrame(wsFrame.getBinaryData().copy()));
```

Comment:
keep chctx and not `chctx.channel()`

Reply:
If I don't call `channel()`, the PONG is never written and the unit tests fail.

Reply:
still we should use the chctx and not the channel for performance reason.

Reply:
if that's never executed why do we have this code at all ?

Reply:
I check this out. Not calling channel works. Using the netty class, `PongWebSocketFrame` works.

Reply:
Yes. I removed the call to `channel` and the tests still passed.  I think the change that fixed this was using the netty class `PongWebSocketFrame` rather than the vertx impl.